### PR TITLE
ENCD-5983-page-and-carts-on-publication-pages

### PR DESCRIPTION
**Pull request by forresttanaka**
Preview demo (fresh demo ready tomorrow):
~https://encd-5983-editable-publication-pv1-fytanaka.demo.encodedcc.org/~

### CODE REVIEW

File: src/encoded/static/components/cart/cart.js [hunk 1]
```diff
@@ -1671,3 +1694,341 @@ Cart.contextTypes = {
...
 export default Cart;
+
+
+/**
+ * Allows the user to select a single assembly for the genome browser.
+ */
+const CartAssemblySelector = ({ selectedAssembly, assemblies, onChangeAssembly }) => {
+    /**
+     * Called when the user makes an assembly selection.
+     * @param {object} event Synthetic even for user's selection
```
Comment (emro) [inline src/encoded/static/components/cart/cart.js]: "event" not "even"

File: src/encoded/static/components/cart/cart.js [hunk 2]
```diff
@@ -1671,3 +1694,341 @@ Cart.contextTypes = {
...
+     */
+    const onChange = (event) => {
+        onChangeAssembly(event.target.value);
+    };
```
Comment (emro) [inline src/encoded/static/components/cart/cart.js]: Is there a reason you have this as a separate function? Couldn't you have something like the following on line 1713?

> onChange={(e) => onChangeAssembly(e.target.value)}

File: src/encoded/static/components/batch_download/cart_static.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        } else {
+            this._downloadOptions.push(
+                {
+                    ...this._downloadOptionsTemplate[this._downloadOptionsTemplateIndex['default-files']],
+                    query: this._defaultFileQueryString,
+                }
+            );
+        }
+
+        // Build the index of all of all the download options.
```
Comment (emro) [inline src/encoded/static/components/batch_download/cart_static.js]: I think there's an extra "all" here

File: src/encoded/static/components/cart/batch_download.js [hunk 4]
```diff
@@ -200,3 +208,147 @@ CartBatchDownload.contextTypes = {
...
+    setSelectedDatasetType,
+}, reactContext) => {
+    const onChange = (event) => {
+        setSelectedDatasetType(event.target.value);
+    };
+
+    return (
+        <div className="cart__batch-download-dataset-type-selector">
+            <>Select download dataset type</>
+            <select value={selectedDatasetType} onChange={onChange}>
```
Comment (emro) [inline src/encoded/static/components/cart/batch_download.js]: I have another comment later on for another instance like this, but I was curious about why you have an onChange function and don't set the dataset type inline.

File: src/encoded/static/components/cart/cart.js [hunk 5]
```diff
@@ -1671,3 +1694,341 @@ Cart.contextTypes = {
...
+    const [processedFiles, setProcessedFiles] = React.useState([]);
+    /** Compiled analyses applicable to the current datasets */
+    const [analyses, setAnalyses] = React.useState([]);
+    /** All types from cart datasets, for downloads */
+    const [datasetTypes, setDatasetTypes] = React.useState([]);
+    /** Current tab's ID */
+    const [displayedTab, setDisplayedTab] = React.useState('browser');
+    /** Currently selected assembly */
+    const [selectedAssembly, setSelectedAssembly] = React.useState('');
+    /** Currently displayed page number for each tab panes; for pagers. */
```
Comment (emro) [inline src/encoded/static/components/cart/cart.js]: each tab pane, not panes

**Review by emro — COMMENTED** [review]
Looks great, there is a lot going on here!

None of my comments are very important, all the code looks good. I think I saw a couple of typos and there was one thing I might have written differently but none of these are required changes.

File: src/encoded/static/components/cart/cart.js (hunk 2 above)
Comment (forresttanaka) [reply src/encoded/static/components/cart/cart.js]: Purely inertia. After years, it still doesn’t occur to me.

File: src/encoded/static/components/cart/cart.js (hunk 5 above)
Comment (forresttanaka) [reply src/encoded/static/components/cart/cart.js]: 👍

File: src/encoded/static/components/batch_download/cart_static.js (hunk 3 above)
Comment (forresttanaka) [reply src/encoded/static/components/batch_download/cart_static.js]: 👍

File: src/encoded/static/components/cart/batch_download.js (hunk 4 above)
Comment (forresttanaka) [reply src/encoded/static/components/cart/batch_download.js]: 👍

**Review by emro — APPROVED** [review]

**Pull request merged**